### DISCUSSION TITLE
How to remove redundant tags

### DISCUSSION THREAD

**Discussion by oeyoews** [TiddlyWiki/TiddlyWiki5#7891] [Q&A] [answered]
```js
$tw.wiki.renderText(
      'text/html',
      'text/vnd.tiddlywiki',
      '<$iconify />',
    );
```

The nodes rendered in this way will always be wrapped by a p tag. Is there any way to remove it besides manually operating the nodes? @Jermolene 

<img width="199" alt="image" src="https://github.com/Jermolene/TiddlyWiki5/assets/72405338/380254e3-31fb-41e3-b763-3965d07e2301">

**Comment by pmario** [comment]
It's not trivial. There is an open PR https://github.com/Jermolene/TiddlyWiki5/pull/7061 which deals with the problem. BUT I did not test the code with v5.3.x yet. 

The main problem will be. It's not really backwards compatible.

**Comment by Jermolene** [comment, marked as answer]
Hi @oeyoews you can parse the text in "inline" mode which will not create any paragraph elements:


```js
$tw.wiki.renderText(
      'text/html',
      'text/vnd.tiddlywiki',
      '<$iconify />',
      {parseAsInline: true}
    );
```
